pykechain/models/widgets/helpers.py

```diff
@@ -107,6 +108,200 @@ def _set_title(
+    :param link: Where the card widget refers to. This can be one of the following:
+        * None (default): no link
+        * task: another KE-chain task, provided as an Activity object or its UUID
+        * String value: URL to a webpage
+        * KE-chain page: built-in KE-chain page of the current scope
+    :type link: basestring or None or bool or KEChainPages
+    :param link_value: Overwrite the default link value (obtained from the type of the link)
+    :type link_value: CardWidgetLinkValue
+    :param link_target: how the link is opened, one of the values of CardWidgetLinkTarget enum.
+    :type link_target: CardWidgetLinkTarget
+
+    :return: tuple of meta
+    :rtype: Tuple[Dict,Text]
+    :raises IllegalArgumentError: When illegal (combination) of arguments are set.
+    """
+    meta['linkTarget'] = check_enum(link_target, LinkTargets, 'link_target')
+
+    from pykechain.models import Activity
+    if isinstance(link, Activity):
+        if link.activity_type == ActivityType.TASK:
+            default_link_value = CardWidgetLinkValue.TASK_LINK
+        else:
+            default_link_value = CardWidgetLinkValue.TREE_VIEW
+
+        meta.update({
+            'customLink': link.id,
+            'showLinkValue': default_link_value,
+        })
+    elif isinstance(link, str) and is_uuid(link):
+        meta.update({
+            'customLink': link,
+            'showLinkValue': CardWidgetLinkValue.TASK_LINK,
+        })
+    elif link is None or link is False:
+        meta.update({
+            'customLink': None,
+            'showLinkValue': CardWidgetLinkValue.NO_LINK,
+        })
+    elif link in KEChainPages.values():
+        meta.update({
+            'customLink': "",
+            'showLinkValue': CardWidgetKEChainPageLink[link],
+        })
+    else:
+        meta.update({
+            'customLink': link,
+            'showLinkValue': CardWidgetLinkValue.EXTERNAL_LINK,
+        })
+
+    if link_value is not None:
+        meta.update({
+            'showLinkValue': check_enum(link_value, CardWidgetLinkValue, 'link_value'),
+        })
+
+    return meta
+
+
+def _set_image(
+        meta: Dict,
+        image: Optional['AttachmentProperty'] = None,
+        image_fit: Optional[Union[Text, ImageFitValue]] = ImageFitValue.CONTAIN,
+        **kwargs
+) -> Tuple[Dict]:
+    """
+    Set the image in the meta based on provided optional custom link.
+
+    This will inject into the meta the `customImage` and `showImageValue` if the `image` is provided
+    as argument, otherwise it will inject 'no link'. It returns the meta definition of the widget.
+
+    :param meta: meta dictionary to augment
+    :type meta: dict
+    :param image: AttachmentProperty providing the source of the image shown in the card widget.
+    :type image: AttachmentProperty or None
+    :param image_fit: how the image on the card widget is displayed
+    :type image_fit: ImageFitValue
+
+    :return: tuple of meta
+    :rtype: Tuple[Dict,Text]
```

Comment:
inconsistent return, meta seems a `Dict` to me
```suggestion
    :rtype: Dict
```

Reply:
Done